src/attestation/Attester.ts

```diff
@@ -126,7 +123,7 @@ export default class Attester implements IAttester {
    *
    * @returns A new [[Accumulator]].
    */
-  public async createAccumulator(): Promise<Accumulator> {
+  public async createAccumulator(): ReturnType<IAttester['createAccumulator']> {
     return new Accumulator(
```

Comment:
Is this idiomatic typescript? I don't like this referencing other types. I feel like it's less readable. Here you would need to have a look at `IAttester` to know what type that is.